src/Arrays/MPIStateArrays.jl

```diff
@@ -295,7 +327,7 @@ function finish_ghost_recv!(Q::MPIStateArray)
   MPI.Waitall!(Q.recvreq)
 
   # copy data to state vectors
-  transferrecvbuf!(Q.device_recvQ, Q.host_recvQ, Q.Q, Q.vmaprecv)
+  transferrecvbuf!(Q.device_recvQ, Q.recvQ, Q.Q, Q.vmaprecv)
```

Comment:
I am not quite sure what `device_recvQ` is being used for, but in the case that `recvQ` resides on the device we should be able to collapse them

Reply:
The array `device_recvQ` is being used to gather the data that needs to be communicated into contiguous chunks for each rank.

Reply:
A long time ago, say ~10 years, I had issues passing a CUDA pinned array into `MPI_Isend`/`MPI_Irecv`.  I guess the MPI installation, which was MVAPICH, was trying to pin the memory region as well and it caused some deadlocking.  Thus, it required double buffering on the host.  I hope this is no longer the case and you can collapse them but haven't tested it.